src/rules/no-separate-loading-state.ts

```
@@ -0,0 +1,136 @@
+
+const LOADING_PATTERNS = [
+  /^is.*Loading$/i,  // isXLoading pattern
+  /^isLoading.+/i,   // isLoadingX pattern
+];
+
+export const noSeparateLoadingState = createRule<[], MessageIds>({
+  name: 'no-separate-loading-state',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow separate loading state variables that track the loading status of other state',
```

ESLint

[error] Insert `⏎·······`

(prettier/prettier)